lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/batch/AzureBatchHelper.java

```diff
@@ -52,16 +54,34 @@
 
   private final BatchClient client;
   private final PoolInformation poolInfo;
+  private CommandBuilder commandBuilder;
+  private final TcpPortProvider portProvider;
+  private final ContainerRegistry containerRegistry;
+  private final ContainerRegistryProvider containerRegistryProvider;
 
   @Inject
   public AzureBatchHelper(
       final AzureBatchFileNames azureBatchFileNames,
       final IAzureBatchCredentialProvider credentialProvider,
+      final TcpPortProvider portProvider,
+      final CommandBuilder commandBuilder,
+      final ContainerRegistryProvider containerRegistryProvider,
       @Parameter(AzureBatchPoolId.class) final String azureBatchPoolId) {
     this.azureBatchFileNames = azureBatchFileNames;
 
     this.client = BatchClient.open(credentialProvider.getCredentials());
     this.poolInfo = new PoolInformation().withPoolId(azureBatchPoolId);
+    this.commandBuilder = commandBuilder;
+    this.containerRegistryProvider = containerRegistryProvider;
+    this.portProvider = portProvider;
+    if (this.containerRegistryProvider.isValid()) {
+      this.containerRegistry = new ContainerRegistry()
+          .withRegistryServer(this.containerRegistryProvider.getContainerRegistryServer())
+          .withUserName(this.containerRegistryProvider.getContainerRegistryUsername())
+          .withPassword(this.containerRegistryProvider.getContainerRegistryPassword());
+    } else {
+      this.containerRegistry = null;
```

Comment:
I agree with Chenxi that this can be removed. Just don't set the value on line 75.

```java
    if (containerRegistryProvider.isValid()) {
      this.containerRegistryProvider = containerRegistryProvider;
      this.containerRegistry = new ContainerRegistry()
          .withRegistryServer(this.containerRegistryProvider.getContainerRegistryServer())
          .withUserName(this.containerRegistryProvider.getContainerRegistryUsername())
          .withPassword(this.containerRegistryProvider.getContainerRegistryPassword());
    }
```